collectors/logsmanagement.plugin/plugin_logsmanagement_web_log.c

```diff
@@ -0,0 +1,995 @@
+        chart_data->num_req_method_orderpatch += p_file_info->parser_metrics->web_log->req_method.orderpatch;
+        p_file_info->parser_metrics->web_log->req_method.orderpatch = 0;
+        chart_data->num_req_method_patch += p_file_info->parser_metrics->web_log->req_method.patch;
+        p_file_info->parser_metrics->web_log->req_method.patch = 0;
+        chart_data->num_req_method_post += p_file_info->parser_metrics->web_log->req_method.post;
+        p_file_info->parser_metrics->web_log->req_method.post = 0;
+        chart_data->num_req_method_pri += p_file_info->parser_metrics->web_log->req_method.pri;
+        p_file_info->parser_metrics->web_log->req_method.pri = 0;
+        chart_data->num_req_method_propfind += p_file_info->parser_metrics->web_log->req_method.propfind;
+        p_file_info->parser_metrics->web_log->req_method.propfind = 0;
+        chart_data->num_req_method_proppatch += p_file_info->parser_metrics->web_log->req_method.proppatch;
+        p_file_info->parser_metrics->web_log->req_method.proppatch = 0;
+        chart_data->num_req_method_put += p_file_info->parser_metrics->web_log->req_method.put;
+        p_file_info->parser_metrics->web_log->req_method.put = 0;
+        chart_data->num_req_method_rebind += p_file_info->parser_metrics->web_log->req_method.rebind;
+        p_file_info->parser_metrics->web_log->req_method.rebind = 0;
+        chart_data->num_req_method_report += p_file_info->parser_metrics->web_log->req_method.report;
+        p_file_info->parser_metrics->web_log->req_method.report = 0;
+        chart_data->num_req_method_search += p_file_info->parser_metrics->web_log->req_method.search;
+        p_file_info->parser_metrics->web_log->req_method.search = 0;
+        chart_data->num_req_method_trace += p_file_info->parser_metrics->web_log->req_method.trace;
+        p_file_info->parser_metrics->web_log->req_method.trace = 0;
+        chart_data->num_req_method_unbind += p_file_info->parser_metrics->web_log->req_method.unbind;
+        p_file_info->parser_metrics->web_log->req_method.unbind = 0;
+        chart_data->num_req_method_uncheckout += p_file_info->parser_metrics->web_log->req_method.uncheckout;
+        p_file_info->parser_metrics->web_log->req_method.uncheckout = 0;
+        chart_data->num_req_method_unlink += p_file_info->parser_metrics->web_log->req_method.unlink;
+        p_file_info->parser_metrics->web_log->req_method.unlink = 0;
+        chart_data->num_req_method_unlock += p_file_info->parser_metrics->web_log->req_method.unlock;
+        p_file_info->parser_metrics->web_log->req_method.unlock = 0;
+        chart_data->num_req_method_update += p_file_info->parser_metrics->web_log->req_method.update;
+        p_file_info->parser_metrics->web_log->req_method.update = 0;
+        chart_data->num_req_method_updateredirectref += p_file_info->parser_metrics->web_log->req_method.updateredirectref;
+        p_file_info->parser_metrics->web_log->req_method.updateredirectref = 0;
```

Comment:
Unless I am missing something, you set all values inside `req_method` to zero, so I will give a suggestion to you, instead to use `var = 0`, use `memset`. This will help your code to be more readable and will cut some lines from the code. :)

Reply:
Of course what I wrote here will apply to next structures.

Reply:
This is also a good point. I should ideally refactor all the web log parsing code, because I wrote it almost 2 years ago and some parts could be reduced to a lot less code.